lang/shared/src/main/scala/com/wavesplatform/lang/package.scala

```diff
@@ -20,4 +20,19 @@ package object lang {
     override def pure[A](x: A): Coeval[A] =
       Coeval.now(x)
   }
+
+  implicit class StringOps(val s: String) extends AnyVal {
+    def isWellFormed: Boolean = {
+      var i = 0
+      while (i < s.length) {
+        val c = s.charAt(i)
+        if (Character.isSurrogate(c)) {
+          if (s.codePointAt(i) == c) return false
+          i += 1
+        }
+        i += 1
+      }
+      true
```

Comment:
How about not using `return` (because it is generally weird in Scala), e.g.
```scala
  def isWellFormed: Boolean = {
    var i = 0
    var wellFormed = true
    while (i < s.length && wellFormed) {
      val c = s.charAt(i)
      if (Character.isSurrogate(c)) {
        if (s.codePointAt(i) == c) wellFormed = false
        else i += 1
      }
      i += 1
    }
    wellFormed
  }
```
?

Reply:
+

Reply:
UP: it works via exceptions in a case of non local returns, but generally it's better to avoid `return`